File: RecycleViewTestApp/app/src/main/java/ru/startandroid/recycleviewtestapp/MainActivity.java
package ru.startandroid.recycleviewtestapp;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

public class MainActivity extends AppCompatActivity {
    RecyclerView rvNumbers;
    NumbersAdapter numbersAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        rvNumbers = (RecyclerView) findViewById(R.id.rvNumbers);

        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        rvNumbers.setLayoutManager(layoutManager);
        rvNumbers.setHasFixedSize(true);

        numbersAdapter = new NumbersAdapter(100, this);
        rvNumbers.setAdapter(numbersAdapter);
    }
}

File: KotlinRecyclerView/app/src/main/java/vk/expencive/kotlinrecyclerview/MainActivity.kt
package vk.expencive.kotlinrecyclerview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private lateinit var blogAdapter: BlogRecyclerAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initRecyclerView()
        addDataSet()
    }

    private fun addDataSet(){
        val data = DataSource.createDataSet()
        blogAdapter.submintList(data)

    }

    private fun initRecyclerView(){
        recycler_view.apply {
            layoutManager = LinearLayoutManager(this@MainActivity)
            val topSpasingDecoration = TopSpasingItemDecoration(30)
            addItemDecoration(topSpasingDecoration)
            blogAdapter = BlogRecyclerAdapter()
            adapter = blogAdapter


        }

//        recycler_view.layoutManager = LinearLayoutManager(this@MainActivity)
//        blogAdapter = BlogRecyclerAdapter()
//        recycler_view.adapter = blogAdapter
    }
}
